anabot/runtime/installation/hub/software_selection.py

```diff
@@ -17,6 +17,7 @@
 from anabot.runtime.errors import TimeoutError
 from anabot.runtime.translate import tr, comps_tr_env, comps_tr_group, comps_tr_env_desc, comps_tr_group_rev, comps_tr_group_desc, comps_tr_env_rev
 from anabot.runtime.installation.common import done_handler
+from anabot.runtime.actionresult import notfound as notfound_new
```

Comment:
Ideally you should use `NotFoundResult` from `anabot.runtime.actionresult` that was created just for such purpose. It lets you specify what couldn't be found (and optionally for which item or where).

Reply:
Sure thing! I am still learning what are the best practices when implementing things to anabot.

This code is inspired by the one used in anabot/runtime/installation/hub/datetime/__init__.py, and I was unaware that there is a better way to do it :)